### PR TITLE
Annotate OpenMode flag classes with bitwise methods

### LINKED ISSUE #52: Flags class should annotate __or__() function
When I write something like
```python
file.open(QtCore.QFile.ReadOnly | QtCore.QFile.Text)
```
It just gives me a warning `Unexpected type: int`.

Flags like `QIODevice.OpenModeFlag` can be or-ed together. So I suggest annotating the `__or__()` function to suppress this warning.

According to the Qt documentation:
"The `OpenMode` type is a typedef for `QFlags<OpenModeFlag>`. It stores an OR combination of `OpenModeFlag` values."
We should annotate `OpenModeFlag` like this:

```python
class QIODevice(QObject):
    class OpenModeFlag(int):
        def __or__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenMode': ...
```

And this should be done for `OpenMode` too.
```python
class QIODevice(QObject):
    ...
    class OpenMode(sip.simplewrapper):
        ...
        def __or__(self, other: typing.Union['QIODevice.OpenModeFlag','QIODevice.OpenMode']) -> 'QIODevice.OpenMode': ...
```

### PR DESCRIPTION
**Pull request by ghost**
Hi there, this is my contribution to this repo. I was chatting with @altendky via Matrix in the `python/typing` channel and I decided I wanted to help out for fun. This issue looked simple enough to do and didn't look like it was being worked on. I haven't tested these changes since I'm new to using Python and mypy.

If all is good then this closes #52

### CODE REVIEW

File: PyQt5-stubs/QtCore.pyi [hunk 1]
```diff
@@ -2277,6 +2288,16 @@ class QIODevice(QObject):
         def __invert__(self) -> 'QIODevice.OpenMode': ...
         def __index__(self) -> int: ...
         def __int__(self) -> int: ...
+        def __and__(self, other: typing.Union['QIODevice.OpenMode', 'QIODevice.OpenModeFlag']) -> 'QIODevice.OpenMode': #type: ignore[override]
+            ...
+        def __iand__(self, other: typing.Union['QIODevice.OpenMode', 'QIODevice.OpenModeFlag']) -> 'QIODevice.OpenMode': #type: ignore[override]
+            ...
+        def __or__(self, other: typing.Union['QIODevice.OpenMode', 'QIODevice.OpenModeFlag']) -> 'QIODevice.OpenMode': #type: ignore[override]
+            ...
+        def __xor__(self, other: typing.Union['QIODevice.OpenMode', 'QIODevice.OpenModeFlag']) -> 'QIODevice.OpenMode': #type: ignore[override]
```
Comment (BryceBeagle) [inline PyQt5-stubs/QtCore.pyi]: Missing `__ixor__`,

File: PyQt5-stubs/QtCore.pyi [hunk 2]
```diff
@@ -2263,6 +2262,18 @@ class QIODevice(QObject):
...
+    class OpenModeFlag(int):
+        def __and__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenModeFlag': #type: ignore[override]
+            ...
+        def __iand__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenModeFlag': #type: ignore[override]
+            ...
+        def __or__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenModeFlag': #type: ignore[override]
+            ...
+        def __xor__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenModeFlag': #type: ignore[override]
+            ...
+        def __ior__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenModeFlag': #type: ignore[override]
```
Comment (BryceBeagle) [inline PyQt5-stubs/QtCore.pyi]: `__ior__` does not exist. `int`s are immutable

File: PyQt5-stubs/QtCore.pyi [hunk 3]
```diff
@@ -2263,6 +2262,18 @@ class QIODevice(QObject):
     NewOnly = ... # type: 'QIODevice.OpenModeFlag'
     ExistingOnly = ... # type: 'QIODevice.OpenModeFlag'
 
+    class OpenModeFlag(int):
+        def __and__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenModeFlag': #type: ignore[override]
+            ...
+        def __iand__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenModeFlag': #type: ignore[override]
```
Comment (BryceBeagle) [inline PyQt5-stubs/QtCore.pyi]: `__iand__` does not exist. `int`s are immutable.

**Review by BryceBeagle — CHANGES_REQUESTED** [review]

**Comment by BryceBeagle** [comment]
Thanks for the contribution @keysemble . Just a few easy to fix issues 🙂

**Comment by BryceBeagle** [comment]
Also, @altendky are we wanting to be picky and keep upstream style? Specifically, upstream puts the `...` at the end of the line always. Do we want to keep that consistent?

**Comment by ghost** [comment]
> Thanks for the contribution @keysemble . Just a few easy to fix issues

No problem. I will get to it now. Been a busy day for me today.

**Review by BryceBeagle — CHANGES_REQUESTED** [review]
It looks like I wasn't clear enough with my change requests, sorry.
`OpenMode.__ixor__` was missing, but the other "in-place" operators for it were correct
`OpenModeFlag` inherits from `int`, which is immutable. It shouldn't have any in-place operators

**Comment by ghost** [comment]
Gotcha. No worries and thanks. Do you mind if I force push with the correct fix?

**Comment by BryceBeagle** [comment]
Go ahead :+1:

**Comment by altendky** [comment]
> Also, @altendky are we wanting to be picky and keep upstream style? Specifically, upstream puts the `...` at the end of the line always. Do we want to keep that consistent?

I dunno...  I would tend to keep upstream style, yeah, but I have some hesitation to require things we can't automatically complain about in CI.  Maybe do for now and hope we get around to finding a helper for this.

@keysemble, my personal preference is to avoid modifying history.  But, opinions definitely vary on this.  `:]`  Also, thanks for helping out!

File: PyQt5-stubs/QtCore.pyi [hunk 4]
```diff
@@ -2263,14 +2263,9 @@ class QIODevice(QObject):
...
     class OpenModeFlag(int):
-        def __and__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenModeFlag': #type: ignore[override]
-            ...
-        def __or__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenModeFlag': #type: ignore[override]
-            ...
-        def __xor__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenModeFlag': #type: ignore[override]
-            ...
-        def __ixor__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenModeFlag': #type: ignore[override]
-            ...
+        def __and__(self, other: typing.Union['QIODevice.OpenModeFlag', 'QIODevice.OpenMode']) -> 'QIODevice.OpenModeFlag': #type: ignore[override] ...
```
Comment (BryceBeagle) [inline PyQt5-stubs/QtCore.pyi]: Looks like the `...`s are after the `#comment`, which is why CI is failing due to invalid syntax

**Review by BryceBeagle — CHANGES_REQUESTED** [review]

File: PyQt5-stubs/QtCore.pyi (hunk 4 above)
Comment (ghost) [reply PyQt5-stubs/QtCore.pyi]: Sorry for the trouble. Fixed. I was being lazy by using Vim's `J` binding for joining both lines.

Comment (BryceBeagle) [reply PyQt5-stubs/QtCore.pyi]: No worries :slightly_smiling_face:

**Review by BryceBeagle — APPROVED** [review]

**Comment by altendky** [comment]
@BryceBeagle are you looking for me to review as well?  I can, but I don't have any particular expectation of that.

**Comment by BryceBeagle** [comment]
Oh yeah I can merge it. Forgot I had that ability now.

**Pull request merged**